presto-main/src/main/java/io/prestosql/sql/gen/RowExpressionCompiler.java

```diff
@@ -234,10 +239,20 @@ public BytecodeNode visitLambda(LambdaDefinitionExpression lambda, Context conte
         @Override
         public BytecodeNode visitVariableReference(VariableReferenceExpression reference, Context context)
         {
+            if (reference.getName().startsWith(TEMP_PREFIX)) {
+                return context.getScope().getTempVariable(reference.getName().substring(TEMP_PREFIX.length()));
+            }
             return fieldReferenceCompiler.visitVariableReference(reference, context.getScope());
         }
     }
 
+    private static final String TEMP_PREFIX = "$$TEMP$$";
+
+    public static VariableReferenceExpression createTempVariableReferenceExpression(Variable variable, Type type)
```

Comment:
was it never needed before?

Reply:
No this is only needed because we are creating a temp variable we want to refer to in another expression.